bootstrap/018_warehouse_updates.sql

```diff
@@ -61,10 +61,12 @@ BEGIN
             newest_completed := (SELECT greatest(coalesce(max(SESSION_END), 0::TIMESTAMP), :newest_completed) FROM RAW_WH_EVT WHERE NOT INCOMPLETE);
             let run_id timestamp := (SELECT run_id FROM RAW_WH_EVT limit 1);
             TRUNCATE TABLE INTERNAL_REPORTING_MV.CLUSTER_AND_WAREHOUSE_SESSIONS_COMPLETE_AND_DAILY_INCOMPLETE;
-            insert into INTERNAL_REPORTING_MV.CLUSTER_AND_WAREHOUSE_SESSIONS_COMPLETE_AND_DAILY_INCOMPLETE SELECT * FROM RAW_WH_EVT WHERE INCOMPLETE OR SESSION_END = :newest_completed;
-            let new_INCOMPLETE number := (select * from TABLE(RESULT_SCAN(LAST_QUERY_ID())));
-            insert into INTERNAL_REPORTING_MV.CLUSTER_AND_WAREHOUSE_SESSIONS_COMPLETE_AND_DAILY select * from RAW_WH_EVT WHERE NOT INCOMPLETE AND SESSION_END <> :newest_completed;
-            let new_closed number := (select * from TABLE(RESULT_SCAN(LAST_QUERY_ID())));
+            let where_clause varchar := (select 'INCOMPLETE OR session_end = to_timestamp_ltz(\'' || :newest_completed || '\')');
```

Comment:
Doesn't need to be a select query, does it? Couldn't it just be

```
let where_clause varchar := 'INCOMPLETE OR session_end = to_timestamp_ltz(\'' || :newest_completed || '\')';
```

Reply:
possibly, the rules of snowsql confuse me. If it doesn't need the select then snowflake will optimise it away

Reply:
fair point.